lib/net/make_net_client.ts

```diff
@@ -26,11 +26,11 @@ export const makeNetClient = (cfg: Net.ClientConfig, logger?: Logger): Net.Clien
           url: _url.toString(),
           headers: {
             ...cfg.headers,
-            ...opts.headers,
+            ...opts?.headers,
```

Comment:
this will break in runtime if `opts` is nullish. we should probably construct the entire request config object in `_opts` on line 19 and pass that to `axios.request`